syntax = "proto3";


message Agent {
    enum Role {
        RECYCLER = 0;
        CONVERTER = 1;
        WASTE_OWNER = 2;
    }

    // The agent's unique public key
    string public_key = 1;

    // A human-readable name identifying the agent
    string name = 2;

    // Whether the Agent is recycler, converter or waste owner
    Role role = 3;

    // Approximately when the agent was registered, as a Unix UTC timestamp
    uint64 timestamp = 4;
}


message AgentContainer {
    repeated Agent entries = 1;
}
